pkg/rancher/migrations.go

```diff
@@ -39,9 +40,21 @@ func forceUpgradeLogout(configMapController controllerv1.ConfigMapController, to
 		}
 	}
 
-	// we only want to migrate if the previous installation was earlier than the migration version
+	// we do not migrate in development environments
+	if rancherversion.Version == "dev" {
+		return nil
+	}
+
+	// if no last migration is found we run
 	if lastMigration, ok := cm.Data[rancherVersionKey]; ok {
-		if lastMigration == "dev" || version.Compare(migrationVersion, lastMigration, "<=") {
+
+		// if a valid sem ver is found we only migrate if the version is less than the target version
+		if semver.IsValid(lastMigration) && semver.IsValid(rancherversion.Version) && version.Compare(migrationVersion, lastMigration, "<") {
```

Comment:
Could still use `<=` here. If versions are the same, we don't migrate. No need to leave it to the next step.